Src/Eiffel/eiffel/byte_code/visitor/assertion_breakable_slot_strategy.e

```diff
@@ -0,0 +1,481 @@
+					l_context.set_assertion_type ({ASSERT_TYPE}.in_loop_invariant)
+					a_node.invariant_part.process (Current)
+				end
+					-- Variant loop byte code
+				if a_node.variant_part /= Void then
+					l_context.set_assertion_type ({ASSERT_TYPE}.in_loop_variant)
+					a_node.variant_part.process (Current)
+				end
+				l_context.set_assertion_type (0)
+			end
+
+			if attached a_node.iteration_exit_condition as e then
+					-- Generate a test for iteration exit condition.
+				e.process (Current)
+			end
+			if attached a_node.stop as s then
+					-- Generate breakpoint slot.
+				increment_counter (a_node)
+					-- Generate a test for loop exit condition.
+				s.process (Current)
+			end
+
+			if a_node.compound /= Void then
+				a_node.compound.process (Current)
+			end
+			if attached a_node.advance_code as a then
+				a.process (Current)
+			end
+
+				-- Save hook context & restore recorded context.
+			body_breakpoint_slot := l_context.get_breakpoint_slot
+			l_context.set_breakpoint_slot (invariant_breakpoint_slot)
+
+			if not (a_node.invariant_part = Void and then a_node.variant_part = Void) then
+					-- Invariant loop byte code
+				if a_node.invariant_part /= Void then
+					l_context.set_assertion_type ({ASSERT_TYPE}.in_loop_invariant)
+					a_node.invariant_part.process (Current)
+				end
+					-- Variant loop byte code
+				if a_node.variant_part /= Void then
+					l_context.set_assertion_type ({ASSERT_TYPE}.in_loop_variant)
+					a_node.variant_part.process (Current)
+				end
+				l_context.set_assertion_type (0)
+			end
+
+				-- Restore hook context
+			l_context.set_breakpoint_slot (body_breakpoint_slot)
+			l_context.set_hidden_code_level (l_old_hidden_code_level)
+		end
+
+	process_loop_expr_b (a_node: LOOP_EXPR_B)
+			-- <Precursor>
+		local
+			invariant_breakpoint_slot: INTEGER
+			body_breakpoint_slot: INTEGER
+			l_context: like context
+			v: detachable VARIANT_B
+			i: detachable BYTE_LIST [BYTE_NODE]
+			l_old_hidden_code_level: INTEGER
+		do
+			l_context := context
+			l_old_hidden_code_level := l_context.hidden_code_level
+			l_context.set_hidden_code_level (0)
+
+				-- Generate loop iteration part.
+--			increment_counter (a_node)`
+			l_context.enter_hidden_code
+			a_node.iteration_code.process (Current)
+			l_context.exit_hidden_code
+
+			v := a_node.variant_code
+			if v /= Void then
+			end
+
+				-- Record context.
+			invariant_breakpoint_slot := l_context.get_breakpoint_slot
+
+			i := a_node.invariant_code
+			if i /= Void or else v /= Void then
+				if i /= Void then
+					l_context.set_assertion_type ({ASSERT_TYPE}.in_loop_invariant)
+					i.process (Current)
+				end
+					-- Variant loop byte code
+				if v /= Void then
+					l_context.set_assertion_type ({ASSERT_TYPE}.in_loop_variant)
+					v.process (Current)
+				end
+				l_context.set_assertion_type (0)
+			end
+				-- Generate byte code for iteration exit condition.
+			a_node.iteration_exit_condition_code.process (Current)
```

Comment:
Should it be hidden?

Reply:
can you develop more ? I don't understand

Reply:
Iteration exit condition is the one that checks `the_cursor.after`. It is not normally shown in the text.